python/examples/core/2-handling-sparsity.py

```diff
@@ -0,0 +1,200 @@
+
+The density :math:`\rho_i^\alpha(r)` associated with all neighbors of species
+:math:`\alpha` of the atom :math:`i` (each neighbor is replaced with a Gaussian function
+centered on the neighbor :math:`g(r_{ij})`) is defined as:
+
+.. math::
+
+    \rho_i^\alpha(r) = \sum_{j \in \text{ neighborhood of i }} g(r_{ij})
+        \delta_{\alpha_j,\alpha}
+
+
+The exact mathematical details above don't matter too much for this tutorial, the main
+point being that this representation treats atomic species as completely independent,
+effectively using the neighbor species :math:`\alpha` for `one-hot encoding`_.
+
+.. _one-hot encoding: https://en.wikipedia.org/wiki/One-hot
+
+.. py:currentmodule:: metatensor
+
+"""
+
+import ase
+import ase.visualize.plot
+import matplotlib.pyplot as plt
+import numpy as np
+
+import metatensor
+
+
+# %%
+#
+# We will work on the radial spectrum representation of three molecules in our system:
+# a carbon monoxide, an oxygen molecule and a nitrogen molecule.
+
+atoms = ase.Atoms(
+    "COO2N2",
+    positions=[(0, 0, 0), (1.2, 0, 0), (0, 6, 0), (1.1, 6, 0), (6, 0, 0), (7.3, 0, 0)],
+)
+
+fig, ax = plt.subplots(figsize=(3, 3))
+ase.visualize.plot.plot_atoms(atoms, ax)
+ax.set_axis_off()
+plt.show()
+
+# %%
+#
+# Sparsity in ``TensorMap``
+# -------------------------
+#
+# The radial spectrum representation has two keys: ``central_species`` indicating the
+# species of the central atom (atom :math:`i` in the equations); and
+# ``neighbor_species`` indicating the species of the neighboring atoms (atom :math:`j`
+# in the equations)
+
+radial_spectrum = metatensor.load("radial-spectrum.npz")
+
+print(radial_spectrum)
+
+# %%
+#
+# This shows the first level of sparsity in ``TensorMap``: block sparsity.
+#
+# Out of all possible combinations of ``central_species`` and ``neighbor_species``, some
+# are missing such as ``central_species=7, neighbor_species=8``. This is because we are
+# using a spherical cutoff of 2.5 Å, and as such there are no oxygen neighbor atoms
+# close enough to the nitrogen centers. This means that all the corresponding radial
+# spectrum coefficients :math:`R_i^\alpha(n)` will be zero (since the neighbor density
+# :math:`\rho_i^\alpha(r)` is zero everywhere).
+#
+# Instead of wasting memory space by storing all of these zeros explicitly, we simply
+# avoid creating the corresponding blocks from the get-go and save a lot of memory!
+
+
+# %%
+#
+# Let's now look at the block containing the representation for oxygen centers and
+# carbon neighbors:
+
+block = radial_spectrum.block(species_center=8, species_neighbor=6)
+
+# %%
+#
+# Naively, this block should contain samples for all oxygen atoms (since
+# ``species_center=8``); in practice we only have a single sample!
+
+print(block.samples)
+
+# %%
+#
+# There is a second level of sparsity happening here, using a format related to
+# `coordinate sparse arrays (COO format) <COO>`_. Since there is only one oxygen atom
```

Comment:
The link is broken. I don't think it works if you use `<COO>` here. sphinx will interpret this as an URL and not as a key.

Reply:
Ah right, there is a separate syntax for this. We could setup a link checker for sphinx to catch these (I find them a bit annoying when working offline, but they might still be quite useful).

Reply:
Yes  maybe we can put in the todo list :-)